lib/fuse_lowlevel.c

```diff
@@ -2907,6 +2907,29 @@ static void fuse_ll_pipe_destructor(void *data)
 	fuse_ll_pipe_free(llp);
 }
 
+static unsigned int get_max_pages(void)
+{
+	char buf[32];
+	int res;
+	int fd;
+
+	fd = open("/proc/sys/fs/fuse/max_pages_limit", O_RDONLY);
+	if (fd < 0)
+		return FUSE_MAX_MAX_PAGES;
+
+	res = read(fd, buf, sizeof(buf) - 1);
+
+	close(fd);
+
+	if (res < 0)
+		return FUSE_MAX_MAX_PAGES;
+
+	buf[res] = '\0';
+
+	res = strtol(buf, NULL, 10);
```

Comment:
That is what I mean with 'inconvenient to use'. Problem is that some architectures have sizeof(int) == 8 - result of LONG_MAX would be positive. I had created a wrapper for strtol in our code that does this check

```
    errno = 0;
    char *endptr;
    long val = strtol(str, &endptr, 10);

    if ((errno == ERANGE && (val == LONG_MAX || val == LONG_MIN)) ||
        (errno == EINVAL) || (errno != 0 && val == 0) || (endptr == str) ||
        (errno == 0 && *endptr != 0))
``

Reply:
Btw, we are testing a new AI tool (since yesterday), it gives me this code suggestion
```
errno = 0;
char *endptr;
long res = strtol(str, &endptr, 10);

if ((errno == ERANGE && (res == LONG_MAX || res == LONG_MIN)) ||
    (errno != 0 && res == 0) ||
    (endptr == str) ||
    (*endptr != '\0' && !isspace((unsigned char)*endptr)))

```

Reply:
(And the people who designed the result of strtol() should be punished by typing this 100 x per day, at least for a month ;) )

Reply:
The highest value that can be set for the `/proc/sys/fs/fuse/max_pages_limit` sysctl value is 65535 and the kernel automatically enforces that the system can't set a value less than or equal to 0 or a non-numerical value. i verified this with
```
$ echo 5a132 | sudo tee /proc/sys/fs/fuse/max_pages_limit
5a132
tee: /proc/sys/fs/fuse/max_pages_limit: Invalid argument
$ echo 0 | sudo tee /proc/sys/fs/fuse/max_pages_limit
0
tee: /proc/sys/fs/fuse/max_pages_limit: Invalid argument
$ echo -12 | sudo tee /proc/sys/fs/fuse/max_pages_limit
-12
tee: /proc/sys/fs/fuse/max_pages_limit: Invalid argument
$ echo 765535 | sudo tee /proc/sys/fs/fuse/max_pages_limit
765535
tee: /proc/sys/fs/fuse/max_pages_limit: Invalid argument
```
Do you think we still need these checks for ERANGE and endptr != '\0'?

Btw I looked at the [glibc implementation of atoi ](https://codebrowser.dev/glibc/glibc/stdlib/atoi.c.html#atoi) and it calls strtol underneath haha 
```
/* Convert a string to an int.  */
int
atoi (const char *nptr)
{
  return (int) strtol (nptr, (char **) NULL, 10);
}
```

Reply:
Yeah, atoi() is not checking for errors.

Reply:
Ok, fine, let's merge it, but I might change it, when we need another char-to-int conversion - from my point of view it makes sense to just add a wrapper for strtol to make it easier to use.

Reply:
Thanks, I'll submit a PR that adds this wrapper and fixes strtol / atoi usages to use the wrapper.